src/substrait/textplan/parser/SubstraitPlanVisitor.cpp

```diff
@@ -0,0 +1,364 @@
+    auto symbol = ANY_CAST(SymbolInfo*, visitSchema_item(item));
+    // TODO -- Implement schemas instead of skipping them.
+    if (symbol == nullptr) {
+      continue;
+    }
+    symbol->location = Location(ctx);
+  }
+
+  return nullptr;
+}
+
+std::any SubstraitPlanVisitor::visitColumn_attribute(
+    SubstraitPlanParser::Column_attributeContext* ctx) {
+  return visitChildren(ctx);
+}
+
+std::any SubstraitPlanVisitor::visitColumn_type(
+    SubstraitPlanParser::Column_typeContext* ctx) {
+  return visitChildren(ctx);
+}
+
+std::any SubstraitPlanVisitor::visitSchema_item(
+    SubstraitPlanParser::Schema_itemContext* ctx) {
+  return symbolTable_->defineSymbol(
+      ctx->id()->getText(),
+      Location(ctx),
+      SymbolType::kSchemaColumn,
+      std::nullopt,
+      ctx->column_type()->getText());
+}
+
+std::any SubstraitPlanVisitor::visitRelation(
+    SubstraitPlanParser::RelationContext* ctx) {
+  auto relType =
+      ANY_CAST(RelationType, visitRelation_type(ctx->relation_type()));
+  if (ctx->relation_ref() == nullptr) {
+    return nullptr;
+  }
+  symbolTable_->defineSymbol(
+      ctx->relation_ref()->id(0)->getText(),
+      Location(ctx),
+      SymbolType::kRelation,
+      relType,
+      ctx);
+  visitRelation_ref(ctx->relation_ref());
+  for (const auto detail : ctx->relation_detail()) {
+    visit(detail);
+  }
+  return nullptr;
+}
+
+std::any SubstraitPlanVisitor::visitRelation_type(
+    SubstraitPlanParser::Relation_typeContext* ctx) {
+  std::string id = ctx->getText();
+  std::transform(id.begin(), id.end(), id.begin(), [](unsigned char c) {
+    return std::tolower(c);
+  });
+  // TODO -- Replace this with a handcrafted function or a trie.
+  if (id == "read") {
+    return RelationType::kRead;
+  } else if (id == "project") {
+    return RelationType::kProject;
+  } else if (id == "join") {
+    return RelationType::kJoin;
+  }
+  this->errorListener_->addError(
+      ctx->getStart(), "Unrecognized relation type: " + ctx->getText());
+  return RelationType::kUnknown;
+}
+
+std::any SubstraitPlanVisitor::visitSource_definition(
+    SubstraitPlanParser::Source_definitionContext* ctx) {
+  symbolTable_->defineSymbol(
+      ctx->read_type()->children[1]->getText(),
+      Location(ctx),
+      SymbolType::kSource,
+      std::nullopt,
+      std::nullopt);
+  return visitChildren(ctx);
+}
+
+std::any SubstraitPlanVisitor::visitConstant(
+    SubstraitPlanParser::ConstantContext* ctx) {
+  return visitChildren(ctx);
+}
+
+std::any SubstraitPlanVisitor::visitColumn_name(
+    SubstraitPlanParser::Column_nameContext* ctx) {
+  return visitChildren(ctx);
+}
+
+std::any SubstraitPlanVisitor::visitSource_reference(
+    SubstraitPlanParser::Source_referenceContext* ctx) {
+  return visitChildren(ctx);
+}
+
+std::any SubstraitPlanVisitor::visitOperation(
+    SubstraitPlanParser::OperationContext* ctx) {
+  // TODO -- Implement this in a second visitor as described below.
```

Comment:
Same here?

Reply:
This section, like most of the file, is not yet implemented.  This comment is merely a reminder about how I plan to solve the issue.  Adding support for functions and relations is part of third PR relating to the parser technology which I'm starting on next week.  Since github doesn't have issue dependencies could we just consider this one to be part of #10?